packages/components/spinner-loader/src/spinner-loader.ts

```diff
@@ -0,0 +1,37 @@
+import { buildProps } from '@puik/utils'
+import type { ExtractPropTypes } from 'vue'
+import type SpinnerLoader from './spinner-loader.vue'
+
+export const spinnerSizes = ['sm', 'md', 'lg']
+export const spinnerColors = ['primary', 'reverse']
+export const spinnerPosition = ['bottom', 'right']
+
+export const spinnerLoaderProps = buildProps({
+  size: {
+    type: String,
```

Comment:
Adding a `PropType` will help reduce error during development

```suggestion
    type: String as PropType<'sm' | 'md' | 'lg'>,
```

The best will be to add an enum for all possible props, and create a type from this enum, to help autocompletion on typescript project

```js
export enum spinnerSizes = {
  SM = 'sm',
  MD = 'md',
  LG = 'lg',
}

export type SpinnerSizeType = `${spinnerSizes}`

export const spinnerLoaderProps = buildProps({
  size: {
    type: String as PropType<SpinnerSizeType>,
    required: false,
    values: spinnerSizes,
    default: 'md',
  },
```

Same for all pre-defined props (colors, size and position)

Reply:
Thx

It's fixed :)